src/TraceEvent/TraceEvent.Tests/TraceEventDispatcherExtensionsTests.cs

```diff
@@ -0,0 +1,489 @@
+            string etlFilePath1 = Path.Combine(UnZippedDataDir, etlFileName1);
+            string etlFilePath2 = Path.Combine(UnZippedDataDir, etlFileName2);
+
+            Dictionary<int, float> processTimes1;
+            using (var source = new ETWTraceEventSource(etlFilePath1))
+            {
+                source.NeedProcesses();
+                source.Process();
+                processTimes1 = source.Processes().ToDictionary(p => p.ProcessID, p => p.CPUMSec);
+            }
+
+            Dictionary<int, float> processTimes2;
+            using (var source = new ETWTraceEventSource(etlFilePath2))
+            {
+                source.NeedProcesses();
+                source.Process();
+                processTimes2 = source.Processes().ToDictionary(p => p.ProcessID, p => p.CPUMSec);
+            }
+
+            using (var source = new ETWTraceEventSource(etlFilePath1))
+            {
+                source.NeedProcesses();
+                source.Process();
+
+                foreach (var process in source.Processes())
+                {
+                    Assert.Equal(processTimes1[process.ProcessID], process.CPUMSec);
+                }
+            }
+
+            using (var source = new ETWTraceEventSource(etlFilePath2))
+            {
+                source.NeedProcesses();
+                source.Process();
+
+                foreach (var process in source.Processes())
+                {
+                    Assert.Equal(processTimes2[process.ProcessID], process.CPUMSec);
+                }
+            }
+        }
+
+        [Fact]
+        [WorkItem(261, "https://github.com/Microsoft/perfview/pull/261")]
+        public void TestNeedProcessesCalledTwice()
+        {
+            Console.WriteLine($"In {nameof(ObservableTests)}.{nameof(TestNeedProcessesCalledTwice)}");
+            PrepareTestData();
+
+            string etlFileName = TestEtlFileNames.First();
+            string etlFilePath = Path.Combine(UnZippedDataDir, etlFileName);
+
+            Dictionary<int, float> processTimes;
+            using (var source = new ETWTraceEventSource(etlFilePath))
+            {
+                source.NeedProcesses();
+                source.Process();
+                processTimes = source.Processes().ToDictionary(p => p.ProcessID, p => p.CPUMSec);
+            }
+
+            using (var source = new ETWTraceEventSource(etlFilePath))
+            {
+                source.NeedProcesses();
+                source.NeedProcesses();
+
+                source.Process();
+
+                foreach (var process in source.Processes())
+                {
+                    Assert.Equal(processTimes[process.ProcessID], process.CPUMSec);
+                }
+            }
+        }
+
+        [Fact(Skip = "https://github.com/Microsoft/perfview/issues/276")]
+        [WorkItem(261, "https://github.com/Microsoft/perfview/pull/261")]
+        public void TestNeedProcessesResetsState()
```

Comment:
:memo: This test was failing prior to the changes made in this pull request, and currently continues to fail.